File: Statistical Learning/01-exercises-package/read_zip.py
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

## Prepare training data
train = pd.read_csv("zip.train", sep = " ", header=None)
print(train.shape)
print(train.isna().sum().sum()) # last columns has only missings
train = train.drop([257], axis=1)

## Prepare test data
test = pd.read_csv("zip.test", sep = " ", header=None)
print(test.isna().sum().sum())
print(test.shape)## V1 is an indicator for the digit, select 2s and 3s
train = train[(train[0] >= 2) & (train[0] <= 3)]
train[0] = train[0] - 2
print(train.shape)
test = test[(test[0] >= 2) & (test[0] <= 3)]
test[0] = test[0] - 2
print(test.shape)

## This is what the images look like:
for i in range(10):
  im = np.array(train[train[0] == i][list(range(2, 258))].iloc[0, :]).reshape(16, 16)
  plt.imshow(np.fliplr(np.rot90(im, 1)), cmap='gray', vmin=0, vmax=63/63)
  plt.axis('off')
  plt.show()
  ## 63 is the minimum number of unique grayscale values observed for any variable 

File: Statistical Learning/01-exercises-package/read_zip.R
## Prepare training data
train <- read.table("/Users/macbookpro/Desktop/learning materials/Statistical Learning/01-exercises-package/zip.train", sep = " ")
dim(train)
colSums(is.na(train)) # last columns has only missings
train <- train[ , -258]

## Prepare test data
test <- read.table("zip.test", sep = " ")
colSums(is.na(test))
dim(test)## V1 is an indicator for the digit, select 2s and 3s
train <- train[train$V1 %in% 2:3, ]
train$V1 <- train$V1 - 2
dim(train)
test <- test[test$V1 %in% 2:3, ]
test$V1 <- test$V1 - 2
dim(test)

## This is what the images look like:
for (i in 0:9) {
  im <- matrix(as.numeric(train[train$V1 == i, 2:257][1, ]), nrow = 16, ncol = 16)
  image(t(apply(-im, 1, rev)), col = gray((0:63)/63), yaxt = "n", xaxt = "n")
  ## 63 is the minimum number of unique grayscale values observed for any variable 
}